src/browser/components/AddressBar/AddressBar.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import classNames from 'classnames';
+import { func, string } from 'prop-types';
+import { noop } from 'lodash';
+
+import ButtonBar from '../ButtonBar';
+import styles from './AddressBar.scss';
+
+const RETURN_KEY = 13;
+
+export default class AddressBar extends React.Component {
+  static propTypes = {
+    className: string,
+    query: string,
+    onQuery: func
+  };
+
+  static defaultProps = {
+    className: null,
+    query: '',
+    onQuery: noop
+  };
+
+  componentDidUpdate(prevProps) {
+    if (this.props.query !== prevProps.query) {
+      this.input.value = this.props.query;
+      this.input.blur();
+    }
+  }
+
+  render() {
+    return (
+      <div className={classNames(styles.addressBar, this.props.className)}>
+        <input
```

Comment:
We have a shared component `Input` - shouldn't this be used instead for consistency?

Reply:
This might be a good idea, yeah

Reply:
I went to change it, but our existing `Input` component is an input wrapped in a label.  The input we display for the address bar is just an input with a placeholder.  I think what we should do as part of a separate PR, is separate `Input` (which just wraps `input`) and `LabelledInput` into two separate components.